upcloud/resource_upcloud_router_test.go

```diff
@@ -90,6 +90,109 @@ func TestAccUpCloudRouter_import(t *testing.T) {
 	})
 }
 
+func TestAccUpCloudRouter_detach(t *testing.T) {
+	var providers []*schema.Provider
+	var router upcloud.Router
+	var network upcloud.Network
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(&providers),
+		CheckDestroy:      testAccCheckRouterNetworkDestroy,
+		Steps: []resource.TestStep{
+			{
+				// first create network and router attached
+				Config: testAccRouterNetworkConfig("testrouter", "testnetwork", true, true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRouterExists("upcloud_router.terraform_test_router", &router),
+					testAccCheckNetworkExists("upcloud_network.terraform_test_network", &network),
+				),
+			},
+			{
+				ResourceName:      "upcloud_router.terraform_test_router",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateCheck: func(states []*terraform.InstanceState) error {
+					for _, s := range states {
+						if s.Attributes["attached_networks.#"] != "1" {
+							return fmt.Errorf("expected 1 network, got %v", s.Attributes["attached_networks.#"])
+						}
+					}
+					return nil
+				},
+			},
```

Comment:
This section can be replaced in the previous checks like
`resource.TestCheckResourceAttr("upcloud_router.testrouter", "name", "testrouter")`
same for number of networks and so one

Reply:
:+1: Excellent, thanks!